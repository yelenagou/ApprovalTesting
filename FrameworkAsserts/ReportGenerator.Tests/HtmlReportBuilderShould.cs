using ApprovalTests.Reporters;
using Xunit;
using ApprovalTests;
using ApprovalTests.Reporters.TestFrameworks;

namespace ReportGenerator.Tests
{
    public class HtmlReportBuilderShould
    {
        [Fact]
        [UseReporter(typeof(FrameworkAssertReporter))]
        public void Build()
        {
            var model = new ReportModel
            {
                Title = "Annual Report",
                ReportLines =
                            {
                                "Line 1xyz",
                                "Line 2",
                                "Line 3",
                                "Line 4",
                                "Line 5"
                            }
            };

            var sut = new HtmlReportBuilder(model);

            string html = sut.Build();

            Approvals.VerifyHtml(html);
        }
    }
}
