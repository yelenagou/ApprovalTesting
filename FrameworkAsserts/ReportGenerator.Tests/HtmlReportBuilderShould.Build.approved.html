<HTML>
  <BODY style="background-color: orange">
    <H1>Annual Report</H1>
    <p>Line 1</p>
    <p>Line 2</p>
    <p>Line 3</p>
    <p>Line 4</p>
    <p>Line 5</p>
  </BODY>
</HTML>